iphone/Maps/Classes/CustomViews/Login/MWMAuthorizationCommon.mm

```diff
@@ -47,31 +47,46 @@ void AuthorizationStoreCredentials(osm::KeySecret const & keySecret)
   [ud synchronize];
 }
 
-BOOL AuthorizationHaveCredentials()
+BOOL AuthorizationHaveOAuth1Credentials()
 {
   NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
   NSString * requestToken = [ud stringForKey:kOSMRequestToken];
   NSString * requestSecret = [ud stringForKey:kOSMRequestSecret];
   return requestToken.length && requestSecret.length;
 }
 
+void AuthorizationClearOAuth1Credentials()
+{
+  NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
+  [ud removeObjectForKey:kOSMRequestToken];
+  [ud removeObjectForKey:kOSMRequestSecret];
+  [ud synchronize];
```

Comment:
This method does nothing and should not be used.
https://developer.apple.com/documentation/foundation/userdefaults/1414005-synchronize

Reply:
Removed calls to `NSUserDefaults.synchronize` method all over the code